.github/workflows/pre-commit.yml

```diff
@@ -0,0 +1,20 @@
+name: pre-commit
```

Comment:
Maybe "lint" is a better name for this action? "pre-commit" is a bit confusing and just an implementation detail of how we're implementing the lint check.